src/pages/RFOX/components/AddressSelection.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  Button,
+  Flex,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  Select,
+  Stack,
+} from '@chakra-ui/react'
+import { useCallback, useMemo, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+export const AddressSelection = () => {
+  const translate = useTranslate()
+  const [isManulAddress, setIsManualAddress] = useState(false)
+
+  const handleToggleInputMethod = useCallback(() => {
+    setIsManualAddress(!isManulAddress)
+  }, [isManulAddress])
+
+  const renderSelection = useMemo(() => {
```

Comment:
This isn't a render fn 
```suggestion
  const selection = useMemo(() => {
```

Reply:
And same for all `renderXYZ`